server/pulp/server/webservices/views/serializers/__init__.py

```diff
@@ -149,6 +154,86 @@ def to_representation(self, instance):
         return representation
 
 
+class ModelSerializer(BaseSerializer):
+    """
+    Base serializer for Mongoengine Documents.
+    """
+
+    def to_representation(self, instance):
+        """
+        Transforms a Mongoengine Document into a serialized dictionary.
+        """
+        document_dict = {}
```

Comment:
I just filed https://pulp.plan.io/issues/1068 in response to this.

The date time conversion is currently being done in `generate_json_respone_with_pulp_encoder`, which makes use of a custom json encoder. If we were to add this functionality to the serializer, we could not stop using our custom json encoder because at that point, the repo dict may have had importers and distributors injected into them (and their non-serialized date times).

I think it would make more sense to first switch to the use a single type of date time (see 1068 for more details), add this conversion to the serializer, and remove the use of pulp's custom json encoder all at once. Otherwise, we are implementing a new way of doing the same thing (pulp will attempt to do it twice) and hope that we get all the uses organically, then remove it. I suspect that in that case, the other date serializers would never be removed and we would just end up with one extra bit of complication.